addons/mail/static/src/core/common/message_model.js

```diff
@@ -287,7 +287,7 @@ export class Message extends Record {
     }
 
     get isHighlightedFromMention() {
-        return this.isSelfMentioned && this.thread?.model === "discuss.channel";
+        return this.isSelfMentioned && this.thread?.channel;
```

Comment:
This will change the API from returning boolean to undefined or channel, which is confusing (and might be less efficient when value changes from false to undefined and code thinks there is an actual value change).

When not directly used inside a condition, it should be wrapped in Boolean (same in all places).

Reply:
I can use `Boolean` to not change the API